tests/test_intercept.js

```diff
@@ -1308,3 +1309,36 @@ test('no new keys were added to the global namespace', t => {
   t.deepEqual(leaks, [], 'No leaks')
   t.end()
 })
+
+// These tests use `http` directly because `got` never calls `http` with the three arg form
+test('first arg as URL instance', t => {
+  const scope = nock('http://example.test')
+    .get('/')
+    .reply(201, 'this is data')
+
+  const req = http.get(new url.URL('http://example.test'))
+
+  req.on('response', () => {
+    scope.done()
+    t.end()
+  })
+})
```

Comment:
I think it could be useful to have a test that the `response` event is emitted. Could you a version of this test to `test_request_overrider` with a title like `'emits the response event'`?

That said, I think the more common way to access the response for the purpose of the `new url.URL()` test would be in a callback, as you've done in the test below.